pallets/network-score/src/tests.rs

```diff
@@ -915,3 +915,84 @@ fn reference_identifier_not_found_test() {
+    let different_creator: SubjectId = SubjectId(AccountId32::new([2u8; 32]));
+
+    let entry = RatingInputEntryOf::<Test> {
+        entity_id,
+        provider_id,
+        total_encoded_rating: 250u64,
+        count_of_txn: 7u64,
+        rating_type: RatingTypeOf::Overall,
+        provider_did: creator.clone(),
+    };
+    let entry_digest = <Test as frame_system::Config>::Hashing::hash(&[&entry.encode()[..]].concat()[..]);
+
+    let raw_space_1 = [2u8; 256].to_vec();
+    let space_digest_1 = <Test as frame_system::Config>::Hashing::hash(&raw_space_1.encode()[..]);
+    let space_id_digest_1 = <Test as frame_system::Config>::Hashing::hash(
+        &[&space_digest_1.encode()[..], &creator.encode()[..]].concat()[..],
+    );
+    let space_id_1: SpaceIdOf = generate_space_id::<Test>(&space_id_digest_1);
+
+    let raw_space_2 = [3u8; 256].to_vec();
+    let space_digest_2 = <Test as frame_system::Config>::Hashing::hash(&raw_space_2.encode()[..]);
+    let space_id_digest_2 = <Test as frame_system::Config>::Hashing::hash(
+        &[&space_digest_2.encode()[..], &creator.encode()[..]].concat()[..],
+    );
+    let space_id_2: SpaceIdOf = generate_space_id::<Test>(&space_id_digest_2);
+
+    // Create an authorization for a different space (space_id_1)
+    let auth_digest_2 = <Test as frame_system::Config>::Hashing::hash(
+        &[&space_id_1.encode()[..], &different_creator.encode()[..], &different_creator.encode()[..]].concat()[..],
+    );
+    let authorization_id_2: AuthorizationIdOf =
+        Ss58Identifier::create_identifier(&auth_digest_2.encode()[..], IdentifierType::Authorization)
+            .unwrap();
+
+    new_test_ext().execute_with(|| {
+        // Create first space
+        assert_ok!(Space::create(
+            DoubleOrigin(author.clone(), creator.clone()).into(),
+            space_digest_1
+        ));
+        assert_ok!(Space::approve(RawOrigin::Root.into(), space_id_1.clone(), 3u64));
+
+        // Create second space
+        assert_ok!(Space::create(
+            DoubleOrigin(author.clone(), creator.clone()).into(),
+            space_digest_2
+        ));
+        assert_ok!(Space::approve(RawOrigin::Root.into(), space_id_2.clone(), 3u64));
+
+        // Attempt to register rating with mismatched space authorization
+        let result = Score::register_rating(
+            DoubleOrigin(author.clone(), creator.clone()).into(),
+            entry.clone(),
+            entry_digest,
+            message_id.clone(),
+            authorization_id_2.clone() 
+        );
+
+        // Print detailed debug information
+        println!("Space ID 1: {:?}", space_id_1);
+        println!("Space ID 2: {:?}", space_id_2);
+        println!("Authorization ID: {:?}", authorization_id_2);
+        println!("Registration Result: {:?}", result);
+
+        // Assert that the registration fails due to mismatched authorization
+        assert!(result.is_err(), "Registration should fail with mismatched authorization");
```

Comment:
Manual assertion is not recommended. 
Look at other test cases how assertion is done for a negative case.

Reply:
```
#[test]
fn test_space_mismatch_in_rating_registration() {
    let creator = DID_00.clone();
    let author = ACCOUNT_00.clone();
    let message_id = BoundedVec::try_from([72u8; 10].to_vec()).unwrap();
    let entity_id = BoundedVec::try_from([73u8; 10].to_vec()).unwrap();
    let provider_id = BoundedVec::try_from([74u8; 10].to_vec()).unwrap();

    // Create first space
    let raw_space_1 = [2u8; 256].to_vec();
    let space_digest_1 = <Test as frame_system::Config>::Hashing::hash(&raw_space_1.encode()[..]);
    let space_id_digest_1 = <Test as frame_system::Config>::Hashing::hash(
        &[&space_digest_1.encode()[..], &creator.encode()[..]].concat()[..],
    );
    let space_id_1: SpaceIdOf = generate_space_id::<Test>(&space_id_digest_1);

    // Create second space
    let raw_space_2 = [3u8; 256].to_vec();
    let space_digest_2 = <Test as frame_system::Config>::Hashing::hash(&raw_space_2.encode()[..]);
    let space_id_digest_2 = <Test as frame_system::Config>::Hashing::hash(
        &[&space_digest_2.encode()[..], &creator.encode()[..]].concat()[..],
    );
    let space_id_2: SpaceIdOf = generate_space_id::<Test>(&space_id_digest_2);

    // Create an authorization for space_id_1
    let auth_digest_1 = <Test as frame_system::Config>::Hashing::hash(
        &[&space_id_1.encode()[..], &creator.encode()[..], &creator.encode()[..]].concat()[..],
    );
    let authorization_id_1: AuthorizationIdOf =
        Ss58Identifier::create_identifier(&auth_digest_1.encode()[..], IdentifierType::Authorization)
            .unwrap();

    // Initialize the rating entry with space_id set to space_id_2 (mismatch)
    let entry = RatingInputEntryOf::<Test> {
        entity_id,
        provider_id,
        total_encoded_rating: 250u64,
        count_of_txn: 7u64,
        rating_type: RatingTypeOf::Overall,
        provider_did: creator.clone(),
        space_id: space_id_2.clone(), // Use space_id_2 here to create a mismatch
    };

    // Calculate entry digest
    let entry_digest = <Test as frame_system::Config>::Hashing::hash(&[&entry.encode()[..]].concat()[..]);

    new_test_ext().execute_with(|| {
        // Create first space
        assert_ok!(Space::create(
            DoubleOrigin(author.clone(), creator.clone()).into(),
            space_digest_1
        ));
        assert_ok!(Space::approve(RawOrigin::Root.into(), space_id_1.clone(), 3u64));

        // Create second space
        assert_ok!(Space::create(
            DoubleOrigin(author.clone(), creator.clone()).into(),
            space_digest_2
        ));
        assert_ok!(Space::approve(RawOrigin::Root.into(), space_id_2.clone(), 3u64));

        // Attempt to register rating with authorization for space_id_1 but using space_id_2
        assert_err!(
            Score::register_rating(
                DoubleOrigin(author.clone(), creator.clone()).into(),
                entry.clone(),
                entry_digest,
                message_id.clone(),
                authorization_id_1.clone()
            ),
            Error::<Test>::SpaceAuthorizationMismatch
        );
    });
}
```

**for this test i am getting error** 
```
missing field `space_id` in initializer of `types::RatingInputEntry<frame_support::BoundedVec<u8, mock::MaxEncodedValueLength>, SubjectId, types::RatingTypeOf>`
  --> pallets\network-score\src\tests.rs:48:14        
   |
48 |     let entry = RatingInputEntryOf::<Test> {     
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `space_id`
```

![image](https://github.com/user-attachments/assets/b9ce7ff1-cba4-4044-821e-3b30f61f2754)

![image](https://github.com/user-attachments/assets/5b972cae-05ff-429e-93e8-4d9556cf771f)


![image](https://github.com/user-attachments/assets/2ca8b3d5-4a23-4987-a01d-5d831c386a50)

**after making all these changes i am getting error! can you help me with this?!**